engines/wintermute/base/scriptables/script_ext_directory.cpp

```diff
@@ -95,13 +98,20 @@ bool SXDirectory::scCallMethod(ScScript *script, ScStack *stack, ScStack *thisSt
 	//////////////////////////////////////////////////////////////////////////
 	else if (strcmp(name, "GetFiles") == 0 || strcmp(name, "GetDirectories") == 0) {
 		stack->correctParams(2);
-		stack->pop()->getString();
+		const char *dirName = stack->pop()->getString();
 		stack->pop()->getString();
 
 		stack->pushInt(0);
 		BaseScriptable *array = makeSXArray(_gameRef, stack);
 
-		warning("Directory.%s is not implemented! Returning empty array...", name);
+		// used in secret scene of "Art of Murder 1: FBI Confidential"
+		if (strcmp(dirName, "X:\\FBI\\data\\scenes\\17-magic\\") == 0 && strcmp(name, "GetDirectories") == 0) {
```

Comment:
This looks like it belongs to another game, e.g. the commit mixes things, but I guess, it is fine.

Reply:
That's correct, 73b669a9d32802cd8efe5d1fe1db7b6e9113c1b1 is unrelated to main subject (except for `error()` changed to `warning()` on `File.Delete()`, which is requied for HeroCraft games). 

Those notes were added while researching on why user profiles were not loaded correctly at HeroCraft games, before noticing that `Game.RegReadString()` is broken, I looked through usage of Directory and File APIs. Since I have decompiled game scripts for almost all games, I decided to run `grep` on them all, not only HeroCraft game scripts only.